g2o/core/base_unary_edge.h

```diff
@@ -27,75 +27,20 @@
 #ifndef G2O_BASE_UNARY_EDGE_H
 #define G2O_BASE_UNARY_EDGE_H
 
-#include <iostream>
-#include <cassert>
-#include <limits>
-
-#include "g2o/config.h"
-#include "g2o/stuff/misc.h"
-#include "base_edge.h"
-#include "robust_kernel.h"
+#include "base_constant_edge.h"
 
 namespace g2o {
 
-
+	// This could be a simple using statement, but in multiple places
+	// _jacobianOplusXi is used.
   template <int D, typename E, typename VertexXi>
-  class BaseUnaryEdge : public BaseEdge<D,E>
+  class BaseUnaryEdge : public BaseConstantEdge<D, E, VertexXi>
```

Comment:
BaseBinaryEdge and BaseUnaryEdge are now pretty useless, this change would allow us to just remove them.

Reply:
Indeed. However, they do now seem to be a part of the API and are used extensively in the types, the examples, and most probably in other projects that depend on g2o. I'm reluctant to break those.